dot/parachain/prospective-parachains/prospective_parachains_test.go

```diff
@@ -1025,3 +1053,135 @@ func TestHandleBacked(
+				mockImplicitlyView := NewMockImplicitView(ctrl)
+				mockImplicitlyView.EXPECT().
+					ActivateLeafFromProspectiveParachains(
+						&backing.BlockInfoProspectiveParachains{
+							Hash:        activeLeafHash,
+							ParentHash:  activeLeafHeader.ParentHash,
+							Number:      parachaintypes.BlockNumber(activeLeafHeader.Number),
+							StorageRoot: activeLeafHeader.StateRoot,
+						},
+						[]*backing.BlockInfoProspectiveParachains{},
+					)
+				mockImplicitlyView.EXPECT().
+					AllAllowedRelayParents().
+					Return([]common.Hash{activeLeafHash})
+
+				mockRuntime := NewMockInstance(ctrl)
+				mockRuntime.EXPECT().
+					ParachainHostClaimQueue().
+					Return(claimQueue, nil)
+				mockRuntime.EXPECT().
+					ParachainHostSessionIndexForChild().
+					Return(parachaintypes.SessionIndex(1), nil)
+				mockRuntime.EXPECT().
+					ParachainHostSchedulingLookAhead().
+					Return(uint32(3), nil)
+
+				constraints := dummyConstraintsV2(0,
+					[]parachaintypes.BlockNumber{0},
+					parachaintypes.HeadData{Data: []byte{0x00}},
+					parachaintypes.ValidationCodeHash{0x00},
+				)
+				mockRuntime.EXPECT().
+					ParachainHostBackingConstraints(parachaintypes.ParaID(1)).
+					Return(constraints, nil)
+
+				mockRuntime.EXPECT().
+					ParachainHostCandidatesPendingAvailability(parachaintypes.ParaID(1)).
+					Return([]parachaintypes.CommittedCandidateReceiptV2{}, nil)
+
+				mockBlockState := NewMockBlockState(ctrl)
+				mockBlockState.EXPECT().
+					GetRuntime(activeLeafHash).
+					Return(mockRuntime, nil)
+
+				mockBlockState.EXPECT().
+					GetHeader(activeLeafHash).
+					Return(activeLeafHeader, nil)
+
+				// mocking fetchAncestry get header call
+				// here we return a different session index
+				// for the parent header
+				mockBlockState.EXPECT().
+					GetHeader(activeLeafHeader.ParentHash).
+					Return(parentHeader, nil)
+
+				innerRuntimeMock := NewMockInstance(ctrl)
+				innerRuntimeMock.EXPECT().
+					ParachainHostSessionIndexForChild().
+					Return(parachaintypes.SessionIndex(0), nil)
+
+				mockBlockState.EXPECT().
+					GetRuntime(activeLeafHeader.ParentHash).
+					Return(innerRuntimeMock, nil)
+
+				subsystemToOverseer := make(chan any)
+
+				pp := NewProspectiveParachains(subsystemToOverseer)
+				pp.blockState = mockBlockState
+				pp.view.implicitView = mockImplicitlyView
+
+				msg := parachaintypes.ActiveLeavesUpdateSignal{
+					Activated: &parachaintypes.ActivatedLeaf{
+						Hash:   activeLeafHash,
+						Number: uint32(activeLeafHeader.Number),
+					},
+				}
+
+				return pp, msg
+			},
+		},
+	}
+
+	for tname, tt := range cases {
+		tt := tt
+		t.Run(tname, func(t *testing.T) {
+			overseerToSubsystem := make(chan any)
+			pp, activeLeafMsg := tt.setup(t)
+
+			ctx, cancel := context.WithCancel(context.Background())
+
+			// Run prospectiveParachains in a separate goroutine
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				pp.Run(ctx, overseerToSubsystem)
```

Comment:
You could call `pp.ProcessActiveLeavesUpdateSignal()` directly and skip the goroutine management.

Reply:
Is there nothing we can assert on to make sure this case was handled correctly? Maybe check that `pp.view` was modified?

Reply:
Good call, most of the asserts are on the functions mocks, but I will add checks on the subsystem state too.